src/test/java/com/codeofus/reservationservice/BaseParkingConsumerIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+package com.codeofus.reservationservice;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryRegistry;
+import io.github.resilience4j.retry.event.RetryEvent;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+@AutoConfigureWireMock
+public class BaseParkingConsumerIntegrationTest extends IntegrationTest {
+    protected static String CIRCUIT_BREAKER_NAME = "parking";
+    protected static String RETRY_NAME = "parking";
+    List<RetryEvent> retryEvents = new ArrayList<>();
+
+    @Autowired
+    private CircuitBreakerRegistry registry;
+
+    @Autowired
+    private RetryRegistry retryRegistry;
+
+    @BeforeEach
+    public void setup() {
+        WireMock.reset();
+        transitionToState(CircuitBreaker.State.CLOSED);
+        configureRetry();
+        retryEvents.clear();
+    }
+
+    protected void transitionToState(CircuitBreaker.State state) {
```

Comment:
You'll probably have more than one CircuitBrakers so you might add a name as an argument in future